### DISCUSSION TITLE
Math Component Tutorial 6.2

### DISCUSSION THREAD

**Discussion by MichaelStarks** [nasa/fprime#1307] [Q&A]
I am trying to complete the fpp-check as directed in the Math Component Tutorial but I am getting an error when running the command.
```
fprime-util fpp-check -u unconnected.txt
[WARNING] Failed to find settings file: /home/michael/Documents/SSRL/fprime-dev/fprime/Ref/settings.ini
[INFO] Updating fpp locations file and build cache. This may take some time.
[ERROR] No fpp dependency memo found. Perhaps directory is outside build?
```
I am in the `Ref/Top` directory and I copied the files from the `docs/Tutorials/MathComponent/Top` folder to see if I messed up but I am still getting the error.

**Comment by LeStarch** [comment]
It looks like it has not be built. Have you run:

```
fprime-util generate
fprime-util build
```

**Reply by MichaelStarks** [reply]
Yes, I have run both of the commands and I am getting the same error.
![image](https://user-images.githubusercontent.com/20226747/157143924-62899d8c-425b-4077-97ee-47b2113ac03c.png)

**Reply by Emil808** [reply]
I faced the same problem too in #1255. Pull to make sure you have the latest update, then here are the fpp install instructions [link](https://github.com/fprime-community/fpp/tree/main/compiler#compiler). Hope this helps.

**Reply by MichaelStarks** [reply]
Thanks for the help but that did not solve my problem. I ran `./test` as instructed in the `Readme`and it finished successfully. 
![image](https://user-images.githubusercontent.com/20226747/157574521-a96ad422-108f-4339-9fde-35207293415f.png)
I removed the build directory, regenerated, and rebuilt the project and I got the same error. I have also tried recloning the fprime repo and copying the files from the math tutorial directory. Unfortunately, the result is the same. I am running the command in `fprime/Ref/Top` for all attempts where `fprime/` is the folder made by `git clone`. What is the `fpp dependency memo` that the command is looking for and what step of the build process is supposed to generate it?

Edit: I purposefully tried to run the command before building the workspace and I got the same error. Does that mean that I am having problems building the workspace?

**Reply by LeStarch** [reply]
What is the version/commit hash you are running on? Is it latest devel?  If so, this function may be in flux.  We are working to speed up FPP and as such changing how the FPP information is exported to user run tools.

**Reply by MichaelStarks** [reply]
I am on 83f3bb92b33e9159e0b1184266fbe34f657e9418. I will rerun the tutorial using the release from December.

**Reply by ThibFrgsGmz** [reply]
I also have this issue even after re-running the generate and build.
I have the head version of the project.

**Reply by LeStarch** [reply]
Latest devel does not run properly any command of the form `fprime-util fpp-*`  I am working on a fix, which will ship with the faster FPP setup in the next release.
